pyface/action/listening_action.py

```diff
@@ -125,7 +125,7 @@ def _object_changed(self, old, new):
         for kind in ("enabled", "visible"):
             method = getattr(self, "_%s_update" % kind)
             name = getattr(self, "%s_name" % kind)
-            if old:
+            if old and old is not Undefined:
```

Comment:
This change was a bit strange, but it is necessary because of https://github.com/enthought/pyface/pull/881/files#r571222562 and here:https://github.com/enthought/traits/blob/aece1feca33e51890f9d0054606a7a241cb2776e/traits/has_traits.py#L327 I believe.

This also made me think that maybe `Undefined` should not evaluate to true in if statements like this, but I do not have enough context on its use to say this at all confidently (it just seems like it would be intuitive).

Reply:
it's worth bringing this up on the channel and hearing from the rest of the team.

Reply:
Suggestion from side-discussion: perhaps use `if isinstance(old, HasTraits)` (and the equivalent for `new` below).